### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Kevin-Mena**
### In this Webapp Project, I Completed the following:
✅ Added a home page with a list of countries that could be filtered by some parameters
✅ Added details page for the country details
✅ Added components and functionalities for both pages
✅ Fetched countries' data from an API
✅ Added styling to the components
✅ Added tests to the components
✅ Deployed the application for an online version

### CODE REVIEW

File: src/components/CountriesList.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          </Link>
+          <img src={country.flags.png} alt={country.flags.alt} />
+          <p>{country.name.common}</p>
+          <p>
+            Population :
+            {country.population}
+          </p>
+        </div>
+      ))}
+    </div>
```
Comment (mckent05) [inline src/components/CountriesList.js]: - Good job done so far, but your homepage is not styled according to the design template. Remember it is important that you try to replicate the Figma design as closely as possible to give confidence to recruiters in your ability to produce exact business requirements 🙏 According to the project requirements, you are required to follow the below design guidlines:
-- Typography: font face, size, and weight
-- Layout: composition and space between elements. Please fix the composition and space between your elements 🙏 

Your project:
awtaw4tw

File: src/components/CountriesList.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  return (
+    <div className="container">
+      <h1>WORLD COUNTRIES INFO</h1>
+      <form>
+        <input
+          type="search"
+          placeholder="Search by country name..."
+          value={searchQuery}
+          onChange={handleChange}
+        />
```
Comment (mckent05) [inline src/components/CountriesList.js]: - Your filter component does not work as expected, I tried searching for a country named `Barbados` in your application and got no result. Please fix this 🙏 . I suggest you consider making sure that both your search parameter and the country named being searched for are both in lower or uppercase letters, you can use the `toLowerCase()` method for this 🙏 
eeae43

File: src/redux/detailsSlice.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      ...state,
+      isLoading: false,
+      countriesList: action.payload,
+    }),
+
+    [getCountries.rejected]: (state) => ({ ...state, isLoading: false }),
+  },
+});
+
+export default detailsReducer.reducer;
```
Comment (mckent05) [inline src/redux/detailsSlice.js]: - You seem not to have used this reducer in any of your components, If so, I suggest deleting this file/reducer, as it is important to keep only necessary files in your project. Please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Kevin-Mena, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- All unit tests are passing, well done 👏 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/detailsSlice.js (hunk 3 above)
Comment (Kevin-Mena) [reply src/redux/detailsSlice.js]: Thank you for the suggestions,I'll implement them as required

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @Kevin-Mena ,
# STATUS: APPROVED! 🥇 💯 
done

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**